tests/test_annotation.py

```diff
@@ -35,118 +44,218 @@ def test_box_gt_upload(dataset):
     annotation = BoxAnnotation(**TEST_BOX_ANNOTATIONS[0])
     response = dataset.annotate(annotations=[annotation])
 
-    assert response['dataset_id'] == dataset.id
-    assert response['annotations_processed'] == 1
-    assert response['annotations_ignored'] == 0
+    assert response["dataset_id"] == dataset.id
+    assert response["annotations_processed"] == 1
+    assert response["annotations_ignored"] == 0
 
-    response = dataset.refloc(annotation.reference_id)['annotations']
+    response = dataset.refloc(annotation.reference_id)["annotations"]
     assert len(response) == 1
     response_annotation = response[0]
-    assert_box_annotation_matches_dict(response_annotation, TEST_BOX_ANNOTATIONS[0])
+    assert_box_annotation_matches_dict(
+        response_annotation, TEST_BOX_ANNOTATIONS[0]
+    )
 
 
 def test_polygon_gt_upload(dataset):
     annotation = PolygonAnnotation(**TEST_POLYGON_ANNOTATIONS[0])
     response = dataset.annotate(annotations=[annotation])
 
-    assert response['dataset_id'] == dataset.id
-    assert response['annotations_processed'] == 1
-    assert response['annotations_ignored'] == 0
+    assert response["dataset_id"] == dataset.id
+    assert response["annotations_processed"] == 1
+    assert response["annotations_ignored"] == 0
 
-    response = dataset.refloc(annotation.reference_id)['annotations']
+    response = dataset.refloc(annotation.reference_id)["annotations"]
     assert len(response) == 1
     response_annotation = response[0]
-    assert_polygon_annotation_matches_dict(response_annotation, TEST_POLYGON_ANNOTATIONS[0])
+    assert_polygon_annotation_matches_dict(
+        response_annotation, TEST_POLYGON_ANNOTATIONS[0]
+    )
+
+
+def test_single_semseg_gt_upload(dataset):
+    annotation = SegmentationAnnotation.from_json(
+        TEST_SEGMENTATION_ANNOTATIONS[0]
+    )
+    response = dataset.annotate(annotations=[annotation])
+    assert response["dataset_id"] == dataset.id
+    assert response["annotations_processed"] == 1
+    assert response["annotations_ignored"] == 0
+    # assert_box_annotation_matches_dict(response_annotation, TEST_BOX_ANNOTATIONS[0])
```

Comment:
In investigating this commented out piece of code, I noticed that running

`response = dataset.refloc(annotation.reference_id)["annotations"]`

Does not return semseg annotations. For a dataset item with both semseg and bbox annotations it only returns the bbox ones. We have to update those backend dataset.refloc/iloc/etc methods to also return semseg annotations as part of full semseg support.

Reply:
(And make sure it works for polygon annotations, I haven't tested that)

Reply:
yep this change should be preceded by a PR in the scaleapi repo but should be pretty quick!